### DISCUSSION TITLE
Chaining Visitors

### DISCUSSION THREAD

**Discussion by Oggy16** [hosseinmoein/DataFrame#248] [Q&A] [answered]
Apologies for another question, but I could not see this in the Hello World example or documentation. 

I wanted to know if it is possible to chain visitors? For example, if I want to apply the rate of change visitor to a linear regression visitor result, how could I do that? The examples usually have a single visitor acting directly on a dataframe from what I can see.

**Comment by hosseinmoein** [comment, marked as answer]
I don't have the chaining interface because it is not as efficient in C++. But what you can do is in one call run the visitor and add its result as a new column in the DataFame [load_result_as_column()](https://htmlpreview.github.io/?https://github.com/hosseinmoein/DataFrame/blob/master/docs/HTML/load_result_as_column.html) (given the result of the algo is a vector). And then run the next visitor on the new column.

**Reply by Oggy16** [reply]
Thank you again! I've got it working by loading the result in a column as you had recommended, so one more blockage removed. 

I can see this requires a different way of designing software, but it is proving an experience. I am absolutely fine with these decisions if they improve efficiency, that is the main reason I am learning this library :-)
